### ISSUE TITLE
Make the api key/password parameters runtime parameters in the cloud PR?

### ISSUE THREAD

**Issue by johann-petrak** [GateNLP/cloud-client#4]
This would make it easier to save a prepared pipeline with the cloud PR which can then be used by different users, each using their own key/pw.

For these parameters, the ability to substitute environment variables may also be useful so that there is no risk ever to accidently save or commit credentials with the pipeline. 
One way could be to process the parameter strings of those parameters with 
https://static.javadoc.io/uk.ac.gate/gate-core/8.5.1/gate/Utils.html#replaceVariablesInString-java.lang.String-java.lang.Object...-
before actually using it.

**Comment by ianroberts** [comment]
The trouble with this is that we (deliberately) create the client at `init` time rather than delaying until `execute`, at least in part so that the client object can be shared among duplicates and have more chance of staying within the rate limit.  To make the parameters runtime we'd have to check at every `execute` call whether the parameter value had changed since the last call and re-create the client at that point if necessary.

**Comment by johann-petrak** [comment]
Yes, but it should be doable -- I have faced the same issue many times in my PRs with other kinds of parameters. That is partly why I thought it is necessary to get the controller started etc. callbacks implemented right.

I was always of the opinion that init parameters are not a very good idea in most cases especially for stuff that can be expected needing to get changed in the GUI. Here it is even worse since it makes it hard to keep access credentials safe. 
It would be better at least for now to have a working pattern for doing such things with runtime parameters.
